api/src/test/java/org/apache/iceberg/TestSchema.java

```diff
@@ -18,32 +18,27 @@
  */
 package org.apache.iceberg;
 
+import static org.apache.iceberg.Schema.DEFAULT_VALUES_MIN_FORMAT_VERSION;
+import static org.apache.iceberg.Schema.MIN_FORMAT_VERSIONS;
+import static org.apache.iceberg.TestHelpers.MAX_FORMAT_VERSION;
 import static org.assertj.core.api.Assertions.assertThatCode;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
+import java.util.List;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.types.Type;
 import org.apache.iceberg.types.Types;
-import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.ValueSource;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.FieldSource;
+import org.junit.jupiter.params.provider.MethodSource;
 
 public class TestSchema {
-  private static final Schema TS_NANO_CASES =
-      new Schema(
-          Types.NestedField.required(1, "id", Types.LongType.get()),
-          Types.NestedField.optional(2, "ts", Types.TimestampNanoType.withZone()),
-          Types.NestedField.optional(
-              3, "arr", Types.ListType.ofRequired(4, Types.TimestampNanoType.withoutZone())),
-          Types.NestedField.required(
-              5,
-              "struct",
-              Types.StructType.of(
-                  Types.NestedField.optional(6, "inner_ts", Types.TimestampNanoType.withZone()),
-                  Types.NestedField.required(7, "data", Types.StringType.get()))),
-          Types.NestedField.optional(
-              8,
-              "struct_arr",
-              Types.StructType.of(
-                  Types.NestedField.optional(9, "ts", Types.TimestampNanoType.withoutZone()))));
+
+  private static final List<Type> TESTTYPES =
```

Comment:
```suggestion
  private static final List<Type> TS_NANO_TYPES =
```

Reply:
This is about to have Variant in it as well, so I was keeping it generic.